src/main/java/sg/dex/starfish/impl/url/RemoteHttpAsset.java

```diff
@@ -27,118 +24,140 @@
+    public InputStream getContentStream() {
+        HttpGet httpget = new HttpGet(uri);
+        CloseableHttpResponse response = null;
+        try {
+            response = HTTP.execute(httpget);
+            StatusLine statusLine = response.getStatusLine();
+            int statusCode = statusLine.getStatusCode();
+            if (statusCode == 404) {
+                throw new RemoteException("Asset ID not found at: " + uri);
+            }
+            if (statusCode == 200) {
+                InputStream inputStream = HTTP.getContent(response);
+                return inputStream;
+            } else {
+                throw new HttpResponseException(statusLine.getStatusCode(), statusLine.getReasonPhrase());
+            }
+        } catch (IOException e) {
+            throw new RemoteException(" Getting Remote Asset content failed: ", e);
+        }
+
     }
 
-	private static String buildMetaData( Map<String, Object> meta) {
-		Map<String, Object> ob = new HashMap<>();
-		ob.put(Constant.DATE_CREATED, Instant.now().toString());
-		ob.put(Constant.TYPE, Constant.DATA_SET);
-		ob.put(Constant.CONTENT_TYPE, "application/octet-stream");
-
-		if (meta != null) {
-			for (Map.Entry<String, Object> me : meta.entrySet()) {
-				ob.put(me.getKey(), me.getValue());
-			}
-		}
-
-		return JSON.toString(ob);
-	}
-
-	/**
-	 * Gets raw data corresponding to this Asset
-	 *
-	 * @return An input stream allowing consumption of the asset data
-	 * @throws AuthorizationException if requestor does not have access permission
-	 * @throws StorageException if unable to load the Asset
-	 */
-	@Override
-	public InputStream getContentStream() {
-		HttpGet httpget = new HttpGet(uri);
-		CloseableHttpResponse response = null;
-		try {
-			response = HTTP.execute(httpget);
-			StatusLine statusLine = response.getStatusLine();
-			int statusCode = statusLine.getStatusCode();
-			if (statusCode == 404) {
-				throw new RemoteException("Asset ID not found at: " + uri);
-			}
-			if (statusCode == 200) {
-				InputStream inputStream = HTTP.getContent(response);
-				return inputStream;
-			} else {
-				throw new HttpResponseException(statusLine.getStatusCode(), statusLine.getReasonPhrase());
-			}
-		}
-		catch (IOException e) {
-			throw new RemoteException(" Getting Remote Asset content failed: ", e);
-		}
-
-	}
-
-	/**
-	 * Gets RemoteAsset size
-	 *
-	 * @return size of the RemoteAsset
-	 */
-	@Override
-	public long getContentSize() {
-		try {
-			return getContentStream().available();
-		}
-		catch (IOException e) {
-			throw new GenericException(
-					"Exception occurred  for asset id :" + getAssetID() + " while finding getting the Content size :",
-					e);
-		}
-	}
-
-	@Override
-	public DID getAssetDID() {
-		throw new UnsupportedOperationException("Can't get DID for asset of type " + this.getClass());
-	}
+    /**
+     * Gets RemoteAsset size
+     *
+     * @return size of the RemoteAsset
+     */
+    @Override
+    public long getContentSize() {
```

Comment:
Makes sense to use idiomatic naming that other Java classes (such as InputStream) use for content length. Is it contentSize/contentLength/available ?